### PR TITLE
Inline WASM in `@nucypher/nucypher-core` NPM package

### PR DESCRIPTION
**Pull request by piotr-roslaniec**
**Type of PR:**
- Feature
- Other

**Required reviews:** 
- 2

**What this does:**
- Add a `rollup` based NPM package bundler in the `./nucypher-core-wasm-bundler` directory
- Configures WASM inlining for the `@nucypher/nucypher-core` package. Now, said package can be used without a WASM-aware bundler. 

**Issues fixed/closed:**
- Related: https://github.com/nucypher/nucypher-ts/pull/299
- Created #85 

**Why it's needed:**
- Simplifies consuming `@nucypher/nucypher-core`

**Notes for reviewers:**
- This is a breaking change for `@nucypher/nucypher-core` users who relied on the previous packaging and WASM module instantiating methods.
- The contents of `./nucypher-core-wasm-bundler` could end up in the `./nucypher-core-wasm` directory, however, I think the current packaging setup in `./nucypher-core-wasm` is worth preserving. We could either document the two distinct approaches or merge them. TBD.
- There is a trade-off when using this inlining technique - The base64-encoded WASM is about 34% larger than in the binary form. We can address it in the future by exposing a "slim" build variant of this library using `package.json` exports. Similarly, we could offer a `node`-friendly "slim" build. See `rollup.config.js` for details. However, these two changes require further research that I can't afford now. I opted to ship these changes earlier at a cost of less flexibility for our users.

### CODE REVIEW

**Review by manumonti — APPROVED** [review]
LGTM! 🙌

I think this is a great progress to have a more user-friendly environment to be used by adopters.

**Comment by manumonti** [comment]
> The contents of ./nucypher-core-wasm-bundler could end up in the ./nucypher-core-wasm directory, however, I think the current packaging setup in ./nucypher-core-wasm is worth preserving. We could either document the two distinct approaches or merge them. TBD.

Some thoughts fwiw:

What could be the cost of keeping the two approaches? Does it involve duplicate work when updating the WASM packages?

If you ask me, my opinion is that is better to keep it simple with only one approach. But I’m not sure about the implications of deprecating the current approach for current adopters.

**Comment by piotr-roslaniec** [comment]
@manumonti 
> What could be the cost of keeping the two approaches? Does it involve duplicate work when updating the WASM packages?

We would only use the new approach when publishing an npm package. We could keep the "legacy" approach as a reference for someone wanting to build a non-inlined package.

**Comment by piotr-roslaniec** [comment]
> How much work would it take someone you used the previous packaging process, to use this updated one? Is it relatively simple once the steps are known? New tooling?

It only requires calling `await initialize()` instead of using a bundler.

> Should this process (migration from prior packaging) be documented alongside fresh (only this new) usage?

We can document it in a changelog. `@nucypher/nucypher-core` is an intermediate dependency with no known external users so I don't think this is going to be an issue.

**Comment by piotr-roslaniec** [comment]
I found an issue caused by the changes in this PR: test runners (like `jest`) have issues calling the `initialize` method. I will address it in this PR.

**Comment by piotr-roslaniec** [comment]
Because of the inlining the size of `@nucypher/nucypher-core` increased by roughly ~33%.

image
https://bundlephobia.com/scan-results?packages=@nucypher/nucypher-core@0.13.0-alpha.1

**Comment by jMyles** [comment]
> We can document it in a changelog. `@nucypher/nucypher-core` is an intermediate dependency with no known external users so I don't think this is going to be an issue.

Moreover, it makes sense for this to be our public stance in general with regard to this repo (and perhaps to finally examine whether we want to call it 'core' rather than something like 'nucypher-base').

File: nucypher-core-wasm-bundler/rollup.config.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    },
+  ],
+});
+
+export default [
+  rolls("umd", "fat"),
+  rolls("es", "fat"),
+  rolls("cjs", "fat"),
+  rolls("cjs", "node"),
+  // TODO: Slim build is not supported ATM
```
Comment (cygnusv) [inline nucypher-core-wasm-bundler/rollup.config.js:60]:
```suggestion
  // TODO: Slim build is not supported ATM - See #85 
```

**Review by cygnusv — APPROVED** [review]
🚀

**Review by KPrasch — APPROVED** [review]

**Pull request merged**